concurrency-primer.tex

```diff
@@ -883,6 +883,94 @@ \subsection{Conclusion about lock-free}
 Balancing complexity and performance is essential in concurrency, 
 a domain fraught with challenges.
 
+\subsection{ABA problem}
+We have introduced CAS as one of the read-modify-write operations.
+However, does the target object not changing really mean that no other threads modified it halfway through?
+If the target object is changed to something by other thread and changed back, the result of comparison is still equal.
+Although the target object has indeed been changed, causing the operation not remaining atomic.
+We call this \introduce{ABA problem}.
+Consider the following scenario,
+
+\inputminted{c}{./examples/simple_aba_example.c}
+
+The execution result would be:
+
+\begin{ccode}
```

Comment:
The following is not code. Instead, it is plaintext. Don't annotate with `code`.